[Android] What is the latest supported RN version?

I'm trying to run with the latest RN version 0.80.0 following the initial Android steps [described here](https://cbl-reactnative.dev/StartHere/install#android), but getting this error when `./gradlew assembleDebug`

```sh
> Task :cbl-reactnative:generateCodegenSchemaFromJavaScript
No modules to process in combine-js-to-schema-cli. If this is unexpected, please check if you set up your NativeComponent correctly. See combine-js-to-schema.js for how codegen finds modules.

> Task :cbl-reactnative:compileDebugKotlin FAILED
e: file:///{omitted}/cbl-reactnative/node_modules/cbl-reactnative/android/src/main/java/com/cblreactnative/DataAdapter.kt:127:45 Only safe (?.) or non-null asserted (!!.) calls are allowed on a nullable receiver of type 'ReadableArray?'.
e: file:///{omitted}/cbl-reactnative/node_modules/cbl-reactnative/android/src/main/java/com/cblreactnative/DataAdapter.kt:128:31 Only safe (?.) or non-null asserted (!!.) calls are allowed on a nullable receiver of type 'ReadableArray?'.
e: file:///{omitted}/cbl-reactnative/node_modules/cbl-reactnative/android/src/main/java/com/cblreactnative/DataAdapter.kt:129:62 Argument type mismatch: actual type is 'String?', but 'String' was expected.
e: file:///{omitted}/cbl-reactnative/node_modules/cbl-reactnative/android/src/main/java/com/cblreactnative/DataAdapter.kt:135:45 Only safe (?.) or non-null asserted (!!.) calls are allowed on a nullable receiver of type 'ReadableArray?'.
e: file:///{omitted}/cbl-reactnative/node_modules/cbl-reactnative/android/src/main/java/com/cblreactnative/DataAdapter.kt:136:31 Only safe (?.) or non-null asserted (!!.) calls are allowed on a nullable receiver of type 'ReadableArray?'.
e: file:///{omitted}/cbl-reactnative/node_modules/cbl-reactnative/android/src/main/java/com/cblreactnative/DataAdapter.kt:137:68 Argument type mismatch: actual type is 'String?', but 'String' was expected.
e: file:///{omitted}/cbl-reactnative/node_modules/cbl-reactnative/android/src/main/java/com/cblreactnative/DataAdapter.kt:529:12 Return type mismatch: expected 'Map<String, Any>', actual 'HashMap<String, Any?>'.

[Incubating] Problems report is available at: file:///{omitted}/cbl-reactnative/android/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.
```

### Steps to reproduce
```sh
npx @react-native-community/cli@latest init cbl-reactnative
cd cbl-reactnative
npm install cbl-reactnative
cd android
# [add] apply from: "../node_modules/cbl-reactnative/android/build.gradle"
./gradlew clean
./gradlew assembleDebug # <- throws
```

ps: iOS works

--- a/android/src/main/java/com/cblreactnative/DataAdapter.kt
+++ b/android/src/main/java/com/cblreactnative/DataAdapter.kt
@@ -517,7 +517,8 @@ object DataAdapter {
         if (value is HashMap<*, *> && value["_type"] == "blob") {
           val nestedMap = value["data"] as HashMap<*, *>
           val contentType = nestedMap["contentType"] as String
-          
+          //value["data"] 'should be' an array of integers - need to convert it because React Native serializes it into
+          //an the ArrayList<Double>
           val rawList = nestedMap["data"] as? ArrayList<*>
           val doubleList = rawList?.filterIsInstance<Double>()
             ?.takeIf { it.size == rawList.size } as? ArrayList<Double>

--- a/android/src/main/java/com/cblreactnative/DataAdapter.kt
+++ b/android/src/main/java/com/cblreactnative/DataAdapter.kt
@@ -123,18 +123,26 @@ object DataAdapter {
     indexProperties?.let { ip ->
       if (indexType == "value") {
         for (countValue in 0 until ip.size()) {
-          val arItems = indexProperties.getArray(countValue)
-          for (countArray in 0 until arItems.size()) {
-            val item = arItems.getString(countArray)
-            valueIndexProperties.add(ValueIndexItem.property(item))
+          val arItems = ip.getArray(countValue)
+          arItems?.let { items ->
+            for (countArray in 0 until items.size()) {
+              val item = items.getString(countArray)
+              item?.let { itemValue ->
+                valueIndexProperties.add(ValueIndexItem.property(itemValue))
+              }
+            }
           }
         }
       } else {
         for (countValue in 0 until ip.size()) {
-          val arItems = indexProperties.getArray(countValue)
-          for (countArray in 0 until arItems.size()) {
-            val item = arItems.getString(countArray)
-            fullTextIndexProperties.add(FullTextIndexItem.property(item))
+          val arItems = ip.getArray(countValue)
+          arItems?.let { items ->
+            for (countArray in 0 until items.size()) {
+              val item = items.getString(countArray)
+              item?.let { itemValue ->
+                fullTextIndexProperties.add(FullTextIndexItem.property(itemValue))
+              }
+            }
           }
         }
       }
@@ -502,31 +510,29 @@ object DataAdapter {
    */
   @Throws(Exception::class)
   fun toMap(readableMap: ReadableMap): Map<String, Any> {
-    val map = readableMap.toHashMap()
-    for ((key, value) in map) {
-      if (value is HashMap<*, *>) {
-        if (value.containsKey("_type") && value["_type"] == "blob") {
+    val resultMap = mutableMapOf<String, Any>()
+    
+    for ((key, value) in readableMap.toHashMap()) {
+      if (value != null) {
+        if (value is HashMap<*, *> && value["_type"] == "blob") {
           val nestedMap = value["data"] as HashMap<*, *>
           val contentType = nestedMap["contentType"] as String
-          //value["data"] 'should be' an array of integers - need to convert it because React Native serializes it into
-          //an the ArrayList<Double>
+          
           val rawList = nestedMap["data"] as? ArrayList<*>
-          val doubleList = rawList?.filterIsInstance<Double>()?.takeIf { it.size == rawList.size } as? ArrayList<Double>
-          val intData = doubleList?.map{ it.toInt()}?.toIntArray()
-          if (intData == null) {
-            throw Exception("Error: Invalid blob data")
-          } else {
-            val data = ByteArray(intData.size)
-            for (i in intData.indices) {
-              data[i] = intData[i].toByte()
-            }
-            val blob = Blob(contentType, data)
-            map[key] = blob
-          }
+          val doubleList = rawList?.filterIsInstance<Double>()
+            ?.takeIf { it.size == rawList.size } as? ArrayList<Double>
+          val intData = doubleList?.map { it.toInt() }?.toIntArray()
+            ?: throw Exception("Error: Invalid blob data")
+          
+          val data = ByteArray(intData.size) { i -> intData[i].toByte() }
+          resultMap[key] = Blob(contentType, data)
+        } else {
+          resultMap[key] = value
         }
       }
     }
-    return map
+    
+    return resultMap
   }
 
   private fun parseIsoDate(dateString: String): Date? {

